tests/extensions/default/analyzers/Microsoft.DotNet.UpgradeAssistant.Extensions.Default.Analyzers.Test/AnalyzerTests.cs

```diff
@@ -145,22 +106,76 @@ public class AnalyzerTests
                     new ExpectedDiagnostic("UA0012", new TextSpan(4943, 39)),
                 }
             },
+
+            // TypeUpgradeAnalyzer scenarios
+            {
+                "HelperResultUpgrade",
+                new[]
+                {
+                    new ExpectedDiagnostic("UA0002", new TextSpan(115, 12)),
+                    new ExpectedDiagnostic("UA0002", new TextSpan(136, 32)),
+                    new ExpectedDiagnostic("UA0002", new TextSpan(209, 32)),
+                    new ExpectedDiagnostic("UA0002", new TextSpan(256, 12)),
+                    new ExpectedDiagnostic("UA0002", new TextSpan(320, 16)),
+                }
+            },
+            {
+                "HtmlStringUpgrade",
+                new[]
+                {
+                    new ExpectedDiagnostic("UA0002", new TextSpan(132, 11)),
+                    new ExpectedDiagnostic("UA0002", new TextSpan(182, 10)),
+                    new ExpectedDiagnostic("UA0002", new TextSpan(307, 21)),
+                    new ExpectedDiagnostic("UA0002", new TextSpan(371, 28)),
+                    new ExpectedDiagnostic("UA0002", new TextSpan(445, 13)),
+                    new ExpectedDiagnostic("UA0002", new TextSpan(497, 13))
+                }
+            },
+            {
+                "ResultUpgrade",
+                new[]
+                {
+                    new ExpectedDiagnostic("UA0002", new TextSpan(240, 25)),
+                    new ExpectedDiagnostic("UA0002", new TextSpan(342, 18)),
+                    new ExpectedDiagnostic("UA0002", new TextSpan(382, 14)),
+                    new ExpectedDiagnostic("UA0002", new TextSpan(423, 12)),
+                    new ExpectedDiagnostic("UA0002", new TextSpan(477, 27)),
+                    new ExpectedDiagnostic("UA0002", new TextSpan(513, 25)),
+                    new ExpectedDiagnostic("UA0002", new TextSpan(563, 14)),
+                    new ExpectedDiagnostic("UA0002", new TextSpan(612, 18))
+                }
+            },
+            {
+                "FilterUpgrade",
+                new[]
+                {
+                    new ExpectedDiagnostic("UA0002", new TextSpan(88, 28)),
```

Comment:
always concerned when refactoring has to change test assertions

```
new ExpectedDiagnostic("UA0004", new TextSpan(82, 28))
new ExpectedDiagnostic("UA0004", new TextSpan(156, 36)),
new ExpectedDiagnostic("UA0004", new TextSpan(321, 37)),
```

Reply:
reading further, maybe this was just a drift created by renaming the classes in these test files

Reply:
Yep, that's the culprit. You'll notice the offsets change very predictably by the different in class name length.

Reply:
That looked good to me, glad you named things by scenario it clears up things like the previously named UA0004A